### PR TITLE
Update magick plugin to OpenSSL 1.1.1 API

### PR DESCRIPTION
**Pull request by JosiahWI**
The functions `EVP_MD_CTX_create` and `EVP_MD_CTX_destroy` were renamed to `EVP_MD_CTX_new` and `EVP_MD_CTX_free` in OpenSSL 1.1.0. This renames them in the magick plugin accordingly. This also replaces the individual calls to `EVP_DigestVerifyUpdate` and `EVP_DigestVerifyFinal` with a single call to `EVP_DigestVerify` which is available in OpenSSL 1.1.1.

### CODE REVIEW

**Comment by JosiahWI** [comment]
LSan detected a leak in the magick plugin and the AuTest prefetch_simple failed:

```
   Run: 2-tr: Failed
     Starting TestRun 2-tr : No Issues found - Passed
        Reason: Started!
     Process: Default: Failed
       Test : Checking that ReturnCode == 0 - Passed
          Reason: Returned Value: 0 == 0
       file /tmp/sandbox/prefetch_simple/_output/2-tr-Default/stream.stdout.txt : Checking that /tmp/sandbox/prefetch_simple/_output/2-tr-Default/stream.stdout.txt matches prefetch_simple.gold - Failed
          Reason: File differences
           Gold File : /home/jenkins/workspace/Github_Builds/autest/src/tests/gold_tests/pluginTest/prefetch/prefetch_simple.gold
           Data File : /tmp/sandbox/prefetch_simple/_output/2-tr-Default/stream.stdout.txt
             GET http://domain.in/texts/demo-1.txt HTTP/1.1
             GET http://domain.in/texts/demo-2.txt HTTP/1.1
             GET http://domain.in/texts/demo-3.txt HTTP/1.1
           - GET http://domain.in/texts/demo-4.txt HTTP/1.1
```

**Comment by JosiahWI** [comment]
[approve ci rocky autest]

File: plugins/experimental/magick/magick.cc [hunk 1]
```diff
@@ -226,19 +226,17 @@ verify(const byte *const msg, const size_t mlen, const byte *const sig, const si
...
-
   ERR_clear_error();
 
   {
-    const int rc = EVP_DigestVerifyFinal(evp.context, sig, slen);
-    return 1 == rc;
+    const int rc = EVP_DigestVerify(evp.context, sig, slen, msg, mlen);
+    if (1 == rc) {
+      return true;
+    } else if (0 == rc) {
```
Comment (maskit) [inline plugins/experimental/magick/magick.cc:235]: If we check this case let's print debug log.

Comment (JosiahWI) [reply plugins/experimental/magick/magick.cc]: Is my use of `Dbg` for this purpose good? I wonder whether we should do this for the API call to load the RSA certificate as well. The current implementation does not even handle such a failure in production. It continues to operate without the cert, and will probably cause everything to fail to verify, if it doesn't do something worse.

**Comment by JosiahWI** [comment]
LSan detected a leak in the magick plugin again.

**Comment by JosiahWI** [comment]
[approve ci rocky]

File: plugins/experimental/magick/magick.cc (hunk 1 above)
Comment (JosiahWI) [reply plugins/experimental/magick/magick.cc]: It does debug asserts and it's probably assuming that it should never fail to load its RSA certificate.

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Comment by JosiahWI** [comment]
[approve ci rocky]

**Review by bneradt — COMMENTED** [review]
Looks reasonable to me.

File: plugins/experimental/magick/magick.cc (hunk 1 above)
Comment (maskit) [reply plugins/experimental/magick/magick.cc]: It just looked like doing exactly same thing as the `else` clause doesn't make sense. Current code still makes me wonder why the error is silently ignored in this case. Leaving a note/comment on the source code may be enough. I'm also fine with letting the `else` clause handle alll the error case.

**Review by maskit — APPROVED** [review]

**Review by bneradt — APPROVED** [review]

**Pull request merged**